src/Microsoft.HttpRepl/FileSystem/IFileSystem.cs

```diff
@@ -0,0 +1,17 @@
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.HttpRepl.FileSystem
+{
+    public interface IFileSystem
+    {
+        Stream Create(string path);
```

Comment:
Names of the methods should indicate whether they refer to file or directory operations.

```
CreateFile
DeleteFile
FileExists
etc...
```

I'm surprised you don't have any directory methods in here.  You can't create a file if its parent doesn't exist, is that supposed to be handled implicitly in the IFileSystem implementation, or should it be exposed explicitly from the IFileSystem?
Example:
```
fileSystem.Create(@"C:\doesnotexist\file.txt");
```
Does this throw (because File.Create would), or should the implementation implicitly call `Directory.CreateDirectory(Path.GetDirectoryName(path))`?  Or should IFileSystem expose a CreateDirectory method that must be called first (just like using the System.IO APIs)?

Reply:
For the names, those started off in different classes (one for File, one for Path and one for Directory) but there were so few I just moved them together and forgot to rename them. I'll change that.
For the directory stuff... so far I was just focused on adding what was already used, and the existing program doesn't check the directory. Should I harden those calls in this PR as well?

Reply:
It would make sense to harden them (whether changing RealFileSystem to check first, or adding Directory.Exists/Directory.Create wrappers).  From the perspective of IFileSystem as a standalone component, this is a dangerous cliff to fall off - better to have a more robust component now.

Reply:
I went ahead and made RealFileSystem check first. There's only two cases where this was happening: 

1. WriteAllTextToFile, used when writing to a temp file, and the directory shouldn't ever not exist in that case
2. WriteAllLinesToFile, used when writing to a file that the user explicitly requested it be written to.

I think in the second case it would be a reasonable expectation of the user that we'd automatically create the directory.